static/js/common.js

```diff
@@ -225,18 +225,24 @@ function renderPanelLogsQueryRes(data, panelId, logLinesViewType, res) {
         //for aggs-query and segstats-query
         if (res.measure && (res.qtype === "aggs-query" || res.qtype === "segstats-query")) {
             let columnOrder = []
-            if (res.groupByCols) {
-                columnOrder = _.uniq(_.concat(
-                    res.groupByCols));
-            }
-            if (res.measureFunctions) {
-                columnOrder = _.uniq(_.concat(
-                    columnOrder, res.measureFunctions));
+            if (res.allColumns !=undefined && res.allColumns.length > 0) {
+                columnOrder = res.allColumns
+            }else{
+                if (res.groupByCols) {
+                    columnOrder = _.uniq(_.concat(
+                        res.groupByCols));
+                }
```

Comment:
inside this else and after the if, wouldn't you need the res.MeasureFunctions also to be concataneted?